tensordict/prototype/tensorclass.py

```diff
@@ -27,6 +27,23 @@
 UNION_PATTERN = re.compile(r"Union\[(.*?)\]")
 
 
+class InvalidAttributeName(ValueError):
+    pass
```

Comment:
This might be overkill, but I couldn't see a built-in exception that accurately described the problem of an invalid attribute name. I think `ValueError` is reasonable, because naming attributes in a class definition is equivalent to passing those names as `attrs` argument to some metaclass, but to the user it feels quite different to calling a function with arguments.